monitored-httpclient/src/main/java/com/github/nhenneaux/resilienthttpclient/monitoredclientpool/HttpClientPoolBuilder.java

```diff
@@ -15,8 +17,7 @@
 
     private DnsLookupWrapper dnsLookupWrapper;
     private ScheduledExecutorService scheduledExecutorService;
-    private SingleHostHttpClientBuilder singleHostHttpClientBuilder;
-    private HttpClient httpClient;
+    private Function<InetAddress, SingleHostHttpClientBuilder> singleHostHttpClientBuilder;
```

Comment:
Should not we rename the variable in this case?

Reply:
What do you propose?

Reply:
On the first place I would not use a function, which seems to  me a bit contr-intuitive. 
But if you insist on using it, then probably something like `singleHostHttpClientBuilderFunction`?

Reply:
it's needed if you want to retry on another tcp connection. I updated the variable name.